android/CMakeLists.txt

```diff
@@ -1,27 +1,150 @@
 cmake_minimum_required(VERSION 3.9.0)
 
-set (PACKAGE_NAME "react-native-quick-sqlite")
 set (CMAKE_VERBOSE_MAKEFILE ON)
-set (CMAKE_CXX_STANDARD 11)
+set (CMAKE_CXX_STANDARD 14)
+set (CMAKE_CXX_FLAGS "-DFOLLY_NO_CONFIG=1 -DFOLLY_HAVE_CLOCK_GETTIME=1 -DFOLLY_HAVE_MEMRCHR=1 -DFOLLY_USE_LIBCPP=1 -DFOLLY_MOBILE=1 -DON_ANDROID -DONANDROID -DFOR_HERMES=${FOR_HERMES}")
+
+set (PACKAGE_NAME "react-native-quick-sqlite")
+set (BUILD_DIR ${CMAKE_SOURCE_DIR}/build)
+set (RN_SO_DIR ${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/first-party/react/jni)
+
+if(${REACT_NATIVE_VERSION} LESS 66)
+        set (
+                INCLUDE_JSI_CPP
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi/jsi/jsi.cpp"
+        )
+        set (
+                INCLUDE_JSIDYNAMIC_CPP
+                "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi/jsi/JSIDynamic.cpp"
+        )
+endif()
+
+file (GLOB LIBFBJNI_INCLUDE_DIR "${BUILD_DIR}/fbjni-*-headers.jar/")
 
 include_directories(
   ../cpp
   "${NODE_MODULES_DIR}/react-native/React"
   "${NODE_MODULES_DIR}/react-native/React/Base"
   "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
+  "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
+  "${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/java/com/facebook/react/turbomodule/core/jni"
+  "${NODE_MODULES_DIR}/react-native/ReactCommon"
+  "${NODE_MODULES_DIR}/react-native/ReactCommon/callinvoker"
+  "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi"
+  "${NODE_MODULES_DIR}/hermes-engine/android/include/"
+  ${INCLUDE_JSI_CPP} # only on older RN versions
+  ${INCLUDE_JSIDYNAMIC_CPP} # only on older RN versions
 )
 
-add_library(sequel
+add_library(
+  ${PACKAGE_NAME}
   SHARED
-  "${NODE_MODULES_DIR}/react-native/ReactCommon/jsi/jsi/jsi.cpp"
-  ../cpp/sequel.cpp
-  ../cpp/sequel.h
-  ../cpp/SequelResult.h
-  ../cpp/react-native-quick-sqlite.cpp
-  ../cpp/react-native-quick-sqlite.h
+  ../cpp/sqliteBridge.cpp
+  ../cpp/sqliteBridge.h
+  ../cpp/installer.cpp
+  ../cpp/installer.h
   ../cpp/sqlite3.h
   ../cpp/sqlite3.c
+  ../cpp/JSIHelper.h
+  ../cpp/JSIHelper.cpp
+  ../cpp/ThreadPool.h
+  ../cpp/ThreadPool.cpp
   cpp-adapter.cpp
 )
 
-target_link_libraries(sequel android log)
+# find fbjni package
+file (GLOB LIBFBJNI_INCLUDE_DIR "${BUILD_DIR}/fbjni-*-headers.jar/")
+
+target_include_directories(
+        ${PACKAGE_NAME}
+        PRIVATE
+        # --- fbjni ---
+        "${LIBFBJNI_INCLUDE_DIR}"
+        # --- Third Party (required by RN) ---
+        "${BUILD_DIR}/third-party-ndk/boost"
```

Comment:
I copied this from sergey's repo, but actually I don't know if it is necessary to download install all this stuff, it would be good if we can keep it simple and remove unnecessary dependencies/download

Reply:
> I copied this from sergey's repo, but actually I don't know if it is necessary to download install all this stuff, it would be good if we can keep it simple and remove unnecessary dependencies/download

Do you belive that my yarn cache reaches 160gb after CMakeLists.txt changes?

Reply:
Installing boost and other C++ is no walk in the park. That is why I want to remove them and make sure we only pull what we need.